src/Features/Core/Portable/NavigateTo/INavigateToSearchResult.cs

```diff
@@ -19,8 +21,42 @@ internal interface INavigateToSearchResult
         string Name { get; }
         ImmutableArray<TextSpan> NameMatchSpans { get; }
         string SecondarySort { get; }
-        string Summary { get; }
+        string? Summary { get; }
 
         INavigableItem NavigableItem { get; }
+        ImmutableArray<PatternMatch> Matches { get; }
+    }
+
+    internal static class NavigateToSearchResultHelpers
+    {
+        /// <summary>
+        /// Helper to bridge from old api that only returned one pattern match, to new API which can return many.
+        /// </summary>
+        public static ImmutableArray<PatternMatch> GetMatches(INavigateToSearchResult result)
+        {
+            var patternMatch = new PatternMatch(
+                GetPatternMatchKind(result.MatchKind),
+                punctuationStripped: false,
+                result.IsCaseSensitive,
+                result.NameMatchSpans);
+            return ImmutableArray.Create(patternMatch);
+        }
+
+        private static PatternMatchKind GetPatternMatchKind(NavigateToMatchKind matchKind)
+            => matchKind switch
+            {
+                NavigateToMatchKind.Exact => PatternMatchKind.Exact,
+                NavigateToMatchKind.Prefix => PatternMatchKind.Prefix,
+                NavigateToMatchKind.Substring => PatternMatchKind.NonLowercaseSubstring,
```

Comment:
why do we need `PatternMatchKind`, `NavigateToMatchKind`, and `PatternMatching.PatternMatchKind`?  Is it possible to keep it at our `PatternMatchKind` and then convert to a single editor kind and remove `NavigateToMatchKind`?

Reply:
yes.  this is a very good point.  prior to EA this was really difficult, because these APIs are implemented by our partners.  but with EA we can likely clean this up.  i will try to do so in this PR.

Reply:
Actually, this isn't great *right now* due to still supporting go-to and aiosp at the same time.  When we can move off of go-to this will be easier to make the conversion.  So i'd like to keep as is for now (and i def agree it's not pleasant).  Is that ok with you?

Reply:
sgtm!